notes/lecture26.tex

```diff
@@ -0,0 +1,224 @@
+\section{Primal Dual IPM}
+Previously, we discussed Barrier Methods and the so-called short-step method for solving constrained LPs, and proved that convergence is guaranteed (albeit slow). Herein, we study a Primal-Dual Interior Point method (the so-called "long-step"  path following algorithm), which similarly seeks to approximate points on the central path. Unlike the short-step method, the long-step method considers iterates of Primal-Dual variables and seeks a more aggressive step size so long as it lies within a neighborhood of the central path.
+
+
+
+\subsection{Deriving the dual problem}
+
+Let $x \in \R^n$ be the decision variable, and $A \in \R^{m \times n},b \in \R^m$ and $c \in \R^n$
+\begin{eqnarray}
+\min c^\top x \quad \text{s.t.}\quad Ax = b, x \ge 0 
+\end{eqnarray}
+$\ge$ is pointwise.
+
+Observe that we can always write the constraint as 
+\begin{eqnarray}
+&& \min_{x \ge 0 } c^\top x + \max_{z} z^{\top}(b-Ax) \\
+&=& \min_{x \ge 0} \max_{z} c^\top x + \max_{z} z^{\top}(b-Ax) \\
+&\ge& \max_{z} z^{\top}b - \min_{x \ge 0} (c - A^{\top}z)^{\top}x\\
+&=& \max_{z} z^{\top}b  - \infty \cdot \Ind(A^{\top}z > c)\\
+\end{eqnarray}
+Hence, the dual problem is as follows
+\begin{eqnarray*}
+\max b^\top z \quad \text{s.t.}\quad A^{\top} z \le c
+\end{eqnarray*}
+This is equivalent to 
+\begin{eqnarray}
+\max b^\top z \quad \text{s.t.}\quad A^{\top} z + s = c, \quad s \ge 0
+\end{eqnarray}
+where we introduce the slack variable $s \in \R^n$. If $(x,z,s)$ are only \emph{feasible}, then 
+\begin{eqnarray}
+Ax = b \quad A^{\top} z + s = c \quad x,s \ge 0
+\end{eqnarray}
+Moreover, we can compute that for feasible $(x,z,s)$,
+\begin{eqnarray}
+0 \le \langle x, s \rangle = x^{\top}(c - A^{\top}z) = \langle x, c \rangle - \langle Ax, z \rangle = \langle x, c \rangle - \langle b,z \rangle~.
+\end{eqnarray}
+This is a proof of weak duality, namely that for any feasible $x$ and $z$,
+\begin{eqnarray}
+\langle x, c \rangle  \ge \langle b,z \rangle
+\end{eqnarray}
+and therefore
+\begin{eqnarray}
+\langle x^*, c \rangle  \ge \langle b,z^* \rangle
+\end{eqnarray}
+Moreover, if there exists an feasible $(x^*,z^*,s^*)$, with $\langle x^*, s^* \rangle = 0$ then we have 
+\begin{eqnarray}
+\langle x^*, c \rangle = \langle b^*,z \rangle~,
+\end{eqnarray}
+which is strong duality. 
+
+Duality is also useful to bound the suboptimality gap, since in fact if $(x,z,s)$ is feasible, then
+\begin{eqnarray}
+\langle x, s \rangle = \langle x, c \rangle - \langle b,z \rangle \ge \langle x,c \rangle - \langle x^*, c \rangle = \langle x - x^* , c \rangle
+\end{eqnarray}
+
+\subsection{Primal-Dual iterates along the Central Path}
```

Comment:
Don't capitalize central path.

Reply:
Also, should be "Primal-dual" here, and "primal-dual" if not at the start of the sentence.